### PR TITLE
Replace list_variable_types with list_logical_types

### LINKED ISSUE #1258: Replace list_variable_types with list_logical_types
- To keep breaking changes to a minimum, the `list_variable_types` should return the DataFrame from `list_logical_types` in Woodwork.
- We should also add a `list_logical_types` (which should just return the same thing as Woodwork's `list_logical_types`
- Let's make sure to note this in the Breaking Changes.

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #1258 by replacing `list_variable_types` with `list_logical_types`

### CODE REVIEW

File: featuretools/variable_types/api.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from .utils import (
     find_variable_types,
     graph_variable_types,
+    list_logical_types,
```
Comment (tamargrey) [inline featuretools/variable_types/api.py:5]: Are we eventually going to get rid of the `variable_types` directory? If so, we probably need somewhere else to put these. 

This also makes me wonder if we're going to need equivalents of `graph_variable_types` and `find_variable_types` (@gsheni )? We might be able to make use of the Woodwork type system's parent-child relationships, but idk

Comment (gsheni) [reply featuretools/variable_types/api.py:5]: We can completely remove `find_variable_types` and `graph_variable_types`. With WW we want to move away from the parent-child relationship. 

Yes, we should get rid of the variable types directory.

Comment (jeff-hernandez) [reply featuretools/variable_types/api.py:5]: Do we want users to import `from featuretools.logical_types import list_logical_types`?

Comment (gsheni) [reply featuretools/variable_types/api.py:5]: If we don't need to have a logical_types folder, let's not have it.`from featuretools import list_logical_types` seems good to me.

Comment (jeff-hernandez) [reply featuretools/variable_types/api.py:5]: Do we want to keep the reference in `featuretools.variable_types.utils` for now and make it possible to import from the top-level `from featuretools import list_logical_types`?

Comment (gsheni) [reply featuretools/variable_types/api.py:5]: My vote would be no. There is already going to be lots of breaking changes, and we can just be sure to note them in the release notes, and add it to the documentation page for V1 upgrade.

**Comment by rwedge** [comment]
I think we should throw a warning if they use `list_variable_types` instead of `list_logical_types` letting them know they should use the new function

File: featuretools/variable_types/api.py (hunk 1 above)
Comment (jeff-hernandez) [reply featuretools/variable_types/api.py:5]: Okay, so we want to import `from featuretools import list_logical_types` and wait to see where to move this function when we delete variable types in a future PR right?

Comment (jeff-hernandez) [reply featuretools/variable_types/api.py:5]: I can also just add another import statement to the reference in `featuretools.__init__`

Comment (gsheni) [reply featuretools/variable_types/api.py:5]: @jeff-hernandez its your call. If you want to do it in a followup MR, just be sure to create an issue. If you want to do it in this MR, that is also fine.

Comment (jeff-hernandez) [reply featuretools/variable_types/api.py:5]: Okay, i'll add another import statement at the top-level

**Comment by tamargrey** [comment]
Can we add a test that checks the contents of `list_logical_types`? Maybe just confirming that it matches the dataframe produced by the direct woodwork call?

**Comment by jeff-hernandez** [comment]
@tamargrey sure, where do we want to place this test?

**Comment by tamargrey** [comment]
> @tamargrey sure, where do we want to place this test?

Hmmm I'd say wherever we're planning on putting `list_variable_types` once we remove the variables directory. Might be best to move it there now, and move its test to the corresponding testing directory so we can stick this new test there as well. 

Maybe `gen_utils.py` and `test_gen_utils.py`? If we end up having a series of woodwork-related utils we might end up creating a woodwork-utils file, but for now, gen utils seems the best fit to me.

**Comment by jeff-hernandez** [comment]
@tamargrey I'll put the test in `test_gen_utils.py` for now and we can come back to `list_variable_types` whenever we remove that directory in a separate PR.

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -69,6 +70,8 @@ Breaking Changes
 * ``last_time_index``, ``secondary_time_index``, and ``interesting_values`` are no longer attributes
   of an entityset’s tables that can be accessed directly. Now they must be accessed through the metadata
   of the Woodwork DataFrame, which is a dictionary.
+* The helper function ``list_variable_types`` has been replaced with ``list_logical_types``.
+  To help minimize breaking changes, ``list_variable_types`` will return the same output as ``list_logical_types``.
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:74]: I think I'd tweak this wording a little bit. The first sentence says we replaced a function, but the second indicates the function still exists. Maybe we could say something like this:

The helper function `list_variable_types` will be removed in a future release and replaced by `list_logical_types`. In the meantime, `list_variable_types` will return the same output as `list_logical_types`.

**Review by thehomebrewnerd — APPROVED** [review]
Just one suggestion for the wording of the breaking change, but otherwise looks good to me.

File: docs/source/release_notes.rst (hunk 2 above)
Comment (jeff-hernandez) [reply docs/source/release_notes.rst:74]: fixed

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -69,6 +70,8 @@ Breaking Changes
 * ``last_time_index``, ``secondary_time_index``, and ``interesting_values`` are no longer attributes
   of an entityset’s tables that can be accessed directly. Now they must be accessed through the metadata
   of the Woodwork DataFrame, which is a dictionary.
+* The helper function ``list_variable_types`` will be removed in a future release and replaced by ``list_logical_types``.
```
Comment (gsheni) [inline docs/source/release_notes.rst:73]: We should add a section to `What's New in this Release` that describes the new list logical types and list semantic tags

**Pull request merged**